File: Data Structures & Algorithms/missing-number/submission-2.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int res = 0;
        int i = 0;
        for(i ; i < nums.size(); i++){
            res ^= nums[i];
            res ^= i;
            std:cout << nums[i] << i << res << std::endl;
        }
        std::cout << i << std::endl;
        res ^= i;

        return res;
    }
};

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-2.py
class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        
        if not nums: return True

        n = len(nums)

        if n == 1: return False

        nums_sum = sum(nums)
        
        if nums_sum % 2 == 1: return False
        
        target = nums_sum // 2

        dp = [False] * (target + 1)
        dp[0] = True

        for num in nums:
            for i in range(target,num-1,-1):
                dp[i] = dp[i] or dp[i-num]

        return dp[-1]
